client/components/header/header.js

```diff
@@ -163,32 +166,35 @@ export default class Header extends React.Component {
               } owisp-header-mobile-menu`}
           >
             {links.map((link, index) => {
-              if (isInternalLink(link.url)) {
+              if (link.authenticated === undefined || link.authenticated === isAuthenticated) {
+                if (isInternalLink(link.url)) {
```

Comment:
I would avoid the nested `if`, eg:

```javascript
if(shouldLinkBeShown(link) && isInternalLink(link.url)) {
  // code
}
```

Reply:
Understanding the way this code was written, I think the second if should stay in place.